src/Response.php

```diff
@@ -174,8 +174,7 @@ public function withStatus($code, $reasonPhrase = '')
      */
     private function setStatusCode($code)
     {
-        if (! is_numeric($code)
-            || is_float($code)
+        if (! is_int($code)
```

Comment:
One potential problem: the status code value may be derived from a string. As an example, it may come from the response line itself:

```http
HTTP/1.1 200 OK
```

In that case, we previously allowed doing something like the following:

```php
[$protocolAndVersion, $statusCode, $reasonPhrase] = explode(' ', $statusLine, 3);
```

and passing the values directly to the constructor and/or `withStatus()`.

With this change, the consumer is now forced to cast to `int` _before_ doing so, which becomes a BC break.